datanode/api/trading_data_v2.go

```diff
@@ -149,6 +153,44 @@ func (t *tradingDataServiceV2) ObserveAccounts(req *v2.ObserveAccountsRequest,
 	})
 }
 
+func (t *tradingDataServiceV2) sendAccountsInitialImage(ctx context.Context, req *v2.ObserveAccountsRequest,
+	srv v2.TradingDataService_ObserveAccountsServer,
+) error {
+	filter := entities.AccountFilter{}
+	if req.Asset != "" {
+		filter.AssetID = entities.AssetID(req.Asset)
+	}
+	if req.PartyId != "" {
+		filter.PartyIDs = append(filter.PartyIDs, entities.PartyID(req.PartyId))
+	}
+	if req.MarketId != "" {
+		filter.MarketIDs = append(filter.MarketIDs, entities.MarketID(req.MarketId))
+	}
+	if req.Type != vega.AccountType_ACCOUNT_TYPE_UNSPECIFIED {
+		filter.AccountTypes = append(filter.AccountTypes, req.Type)
+	}
+
+	accounts, pageInfo, err := t.accountService.QueryBalances(ctx, filter, entities.CursorPagination{})
+	if err != nil {
+		return errors.Wrap(err, "fetching account balance initial image")
+	}
+
+	if pageInfo.HasNextPage {
+		return fmt.Errorf("initial image spans multiple pages")
+	}
```

Comment:
How bit is the default page again? We may want to go against this restriction there, or eventually find a way to send all the pages one after the other. Basically put all the stuff in between line 182 and 191 in a `for pageInfo.HasNextPage {...}`

Reply:
So the 'default'  page you get by making an empty `Pagination{}` is infinitely big. There's a different default you get by calling NewDefaultPaginatinon() intended to be used that as a default when people call a GRPC method without specifying, which is limited to 1000 or something

Reply:
@MM0819 - question for you.. Would it be OK to receive the initial image over multiple messages? @jeremyletang and I just discussing the case where the initial image might be larger than can fit in a GRPC message. We've added pagination to all the other endpoints to solve that problem.

It might be the case that because we're only ever getting live orders etc.. it's unlikely we'll be returning enough data for it to be a problem but perhaps it is prudent to design it to cope with that up front.

to be clear, i'm not suggesting changing the format of the result, adding cursors etc.. just to send initial images in batches of a few hundred or something.

Reply:
Hmmm, so there are a few things here. On other exchanges, there is typically a type field telling you how to handle the message (for example, there will be a field saying `type=snapshot` or `type=update`). The way that a snapshot messages are handled is by replacing your entire state. So, what you propose clearly wouldn't work with that design pattern. 

However, I don't know if we differentiate between the message types in the same way. In which case, I think it would work to receive the snapshot over multiple messages, as long as I don't receive any updates between snapshot messages because I might receive updates for orders where I have not yet received the snapshot.

Also, it's helpful to disable any trading strategy from making trading decisions until its state is 100% synced. This means I would need a way to identify when the final snapshot message has been received when multiple snapshot messages are being sent.

Reply:
OK so here's the solution we've come up with: The observe methods now have a response like 
```

message ObservePositionsResponse {
    oneof response {
        // An 'initial image' snapshot containing current positions
        PositionSnapshotPage snapshot = 1;
        // A list of position updates in the last block
        PositionUpdates updates = 2;
    }
}

// 'Initial image' of current positions, may be sent over multiple response messages
message PositionSnapshotPage{
    repeated vega.Position positions = 1;
    bool last_page = 2;
}

// A list of position updates in the last block
message PositionUpdates{
    repeated vega.Position positions = 1;
}
```

So you can tell if it's a snapshot or not by looking at the `oneof` in the response - and the last page of the snapshot will have `last_page=True`